homeassistant/components/vicare/__init__.py

```diff
@@ -145,8 +150,14 @@ async def async_migrate_entities(
 ) -> None:
     """Migrate old entry."""
     gateway_serial: str = device.config.getConfig().serial
-    device_serial: str = device.api.getSerial()
-    new_identifier = f"{gateway_serial}_{device_serial}"
+    device_id = device.config.getId()
+    device_serial: str | None = await hass.async_add_executor_job(
+        get_device_serial, device.api
+    )
+
+    new_identifier = (
+        f"{gateway_serial}_{device_serial if device_serial is not None else device_id}"
```

Comment:
We need to add a test for the new case here.

Reply:
done